jwt/__init__.py

```diff
@@ -38,9 +38,24 @@ class EncodeError(Exception): pass
     'RS256': lambda msg, key, sig: key.verify(hashlib.sha256(msg).digest(), sig, 'sha256'),
     'RS384': lambda msg, key, sig: key.verify(hashlib.sha384(msg).digest(), sig, 'sha384'),
     'RS512': lambda msg, key, sig: key.verify(hashlib.sha512(msg).digest(), sig, 'sha512'),
-    'none': lambda msg, key, sig: True
+    'none': lambda msg, key, sig: check_none_verify(msg, key, sig)
 }
 
+SUPPORTED_ALGOS = tuple(verifying_methods.keys())
+ALLOWED_ALGOS = sorted(SUPPORTED_ALGOS)
+# Remove the 'none' type from our particular usage
+del ALLOWED_ALGOS[len(ALLOWED_ALGOS) - 1] 
```

Comment:
ah, I see that you added a keyword arg to ``decode()`` and also provided a way to set/remove algorithms. Hmm. I guess removing none by default is fine.